src/main/java/bio/terra/cloudfunctions/streaming/GcsBQ.java

```diff
@@ -1,83 +1,91 @@
-    logger.info("Created: " + event.getTimeCreated());
-    logger.info("Updated: " + event.getUpdated());
-    logger.info("Media download link: " + event.getMediaLink());
-    throw new UnsupportedOperationException("Not supported yet.");
-  }
-
-  public static class GCSEvent {
-    String bucket;
-    String name;
-    Long metageneration;
-    Date timeCreated;
-    Date updated;
-
-    public String getBucket() {
-      return bucket;
-    }
-
-    public String getName() {
-      return name;
-    }
-
-    public void setName(String name) {
-      this.name = name;
-    }
-
-    public Long getMetageneration() {
-      return metageneration;
-    }
-
-    public void setMetageneration(Long metageneration) {
-      this.metageneration = metageneration;
-    }
-
-    public Date getTimeCreated() {
-      return timeCreated;
-    }
-
-    public void setTimeCreated(Date timeCreated) {
-      this.timeCreated = timeCreated;
-    }
-
-    public Date getUpdated() {
-      return updated;
-    }
-
-    public void setUpdated(Date updated) {
-      this.updated = updated;
-    }
-
-    public String getMediaLink() {
-      return mediaLink;
-    }
-
-    public void setMediaLink(String mediaLink) {
-      this.mediaLink = mediaLink;
-    }
-
-    public void setBucket(String bucket) {
-      this.bucket = bucket;
+    logger.info(event.toString());
+
+    for (Map.Entry<String, String> entry : System.getenv().entrySet())
+      logger.info("Key = " + entry.getKey() + ", Value = " + entry.getValue());
+
+    String projectId = System.getenv("GCLOUD_PROJECT");
+    String bucketName = event.getBucket();
+    String objectName = event.getName();
+
+    InputStream in = getObjectAsInputStream(projectId, bucketName, objectName);
+
+    // Uncompress .gz as CompressorInputStream
+    CompressorStreamFactory compressor = CompressorStreamFactory.getSingleton();
+    CompressorInputStream uncompressedInputStream =
+        in.markSupported()
+            ? compressor.createCompressorInputStream(in)
+            : compressor.createCompressorInputStream(new BufferedInputStream(in));
+
+    // Untar .tar as ArchiveInputStream
+    ArchiveStreamFactory archiver = new ArchiveStreamFactory();
+    ArchiveInputStream archiveInputStream =
+        uncompressedInputStream.markSupported()
+            ? archiver.createArchiveInputStream(uncompressedInputStream)
+            : archiver.createArchiveInputStream(new BufferedInputStream(uncompressedInputStream));
+
+    // ArchiveInputStream is a special type of InputStream that emits an EOF when it gets to the end
+    // of a file in the archive.
+    // Once it’s done, call getNextEntry to reset the stream and start reading the next file.
+    // When getNextEntry returns null, you’re at the end of the archive.
+    ArchiveEntry archiveEntry;
+    while ((archiveEntry = archiveInputStream.getNextEntry()) != null) {
+      logger.info(archiveEntry.getName() + " " + archiveEntry.getSize() + " bytes");
     }
+  }
```

Comment:
I'd really like to separate the different pieces of code into separate classes, for encapsulation and future-proofing. For instance, we should have one "harness" class for interfacing with Cloud Functions, one or more classes for handling TestRunner files, etc. I've made a rough sketch of this here: https://lucid.app/lucidchart/invitations/accept/inv_fc1efe60-354d-4d0d-a91d-44760463a6df?viewport_loc=-44%2C12%2C2503%2C1291%2C7ezayB~FQlW7. I don't expect this refactoring as part of this PR, but thought I'd mention it.

Reply:
Created story [QA-1523](https://broadworkbench.atlassian.net/browse/QA-1523) to work on the comments.